pkg/coreapi/graph/resolvers/event_v2.go

```diff
@@ -25,7 +26,67 @@ func (qr *queryResolver) EventV2(ctx context.Context, id ulid.ULID) (*models.Eve
 	return cqrsEventToGQLEvent(event), nil
 }
 
-func (e eventV2Resolver) Raw(ctx context.Context, obj *models.EventV2) (string, error) {
+func (er eventV2Resolver) Runs(ctx context.Context, obj *models.EventV2) ([]*models.FunctionRunV2, error) {
+	// convert cqrs TraceRun to FunctionRunV2
+
+	// This is an N+1, currently also an N+1 on cloud in the form of multiple calls to the metrics service
+	// we cannot currently use a data loader due to current db schema and https://github.com/sqlc-dev/sqlc/issues/1830
+	traceRuns, err := er.Data.GetTraceRunsByTriggerID(ctx, obj.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	functionRuns := make([]*models.FunctionRunV2, len(traceRuns))
+	for i, r := range traceRuns {
+		// TODO dedupe cqrs.TraceRun to models.FunctionRunV2 transformation
+		var (
+			started   *time.Time
+			ended     *time.Time
+			sourceID  *string
+			output    *string
+			batchTime *time.Time
+		)
+
+		if r.StartedAt.UnixMilli() > 0 {
+			started = &r.StartedAt
+		}
+		if r.EndedAt.UnixMilli() > 0 {
+			ended = &r.EndedAt
+		}
+		if len(r.SourceID) > 0 {
+			sourceID = &r.SourceID
+		}
+		if len(r.Output) > 0 {
+			s := string(r.Output)
+			output = &s
+		}
+		runID := ulid.MustParse(r.RunID)
+		status, err := models.ToFunctionRunStatus(r.Status)
+		if err != nil {
+			continue
```

Comment:
This can lead to a sparse slice. Is that a problem?

Reply:
hm true, we'd return the zero struct of `FunctionRunV2` here but I think it'd be better to change it to not return a sparse slice

Reply:
ok changed it to initialize with 0 len and the same capacity and append instead